File: Arrays & Hashing/3. Two Sum/solution.scala
object Solution {
  def main(args: Array[String]): Unit = {
    val nums = Array(2, 7, 11, 5)
    val target = 9
    val result = twoSum(nums, target)
    println(result)
  }

  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    for (i <- 0 to nums.length - 1) {
      for (j <- i + 1 to nums.length - 1) {
        if (nums(i) + nums(j) == target) {
          return Array(i, j)
        }
      }
    }
    return Array(1)
  }
}

File: Arrays & Hashing/3. Two Sum/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  result = []
  nums_hash = {}

  nums.each_with_index do |num, index|
    previous_index = nums_hash[(target - num)]

    if !previous_index.nil?
      result << previous_index
      result << index
      break
    end

    nums_hash[num] =  index
  end

  return result;
end
